src/azure-cli/azure/cli/command_modules/sql/custom.py

```diff
@@ -2211,6 +2211,16 @@ def managed_instance_update(
     instance.proxy_override = (
         proxy_override or instance.proxy_override)
 
+    instance.sku.name = None
```

Comment:
tiny nit: consider commenting the reason why you do this, since it would not be obvious when first reading